### PR TITLE
Remove debug information when debugging

### LINKED ISSUE #10756: Remove debug information when debugging
This is noisy and useless. we should only print that when R_DEBUG=1 env is set

screen shot 2018-07-17 at 12 18 40

### PR DESCRIPTION
**Pull request by pawlos**
Added a check to see if the R_DEBUG env variable is set before printing debug information. This closes #10756

### CODE REVIEW

File: binr/radare2/radare2.c [hunk 1]
```diff
@@ -1191,17 +1191,20 @@ int main(int argc, char **argv, char **envp) {
 			/* load symbols when doing r2 -d ls */
 			// NOTE: the baddr is redefined to support PIE/ASLR
 			baddr = r_debug_get_baddr (r.dbg, pfile);
-			if (baddr != UT64_MAX && baddr != 0) {
+			char *env_debug = r_sys_getenv ("R_DEBUG");
```
Comment (ret2libc) [inline binr/radare2/radare2.c]: `env_debug` should be freed

File: libr/debug/debug.c [hunk 2]
```diff
@@ -575,7 +575,8 @@ R_API bool r_debug_select(RDebug *dbg, int pid, int tid) {
 		tid = pid;
 	}
 	if (pid != -1 && tid != -1) {
-		if (pid != dbg->pid || tid != dbg->tid) {
+		char *env_debug = r_sys_getenv ("R_DEBUG");
```
[the remark this answers is not in the document]
Comment (ret2libc) [reply libr/debug/debug.c]: I agree with @Maijin . `dbg.verbose` seems more in line with how we deal with things in other parts of the code as well (e.g. esil.verbose, bin.verbose, etc.)

[the remark this answers is not in the document]
Comment (ret2libc) [reply libr/debug/debug.c]: btw, `env_debug` should be freed here as well.

**Comment by radare** [comment]
R_DEBUG env is used for debugging r2 internals, like getting error messages when trying to load plugins and so on. Use eval config vars for this

**Comment by radare** [comment]
what about using the r_log api instead

**Comment by pawlos** [comment]
I'll have a look at it.

File: libr/debug/debug.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (radare) [reply libr/debug/debug.c]: dont use getenv in rdebug api. just hold a dbg->verbose variable and set it from core when the config var changes. also r_sys_getenv() returns a new string in the heap that you should free

File: binr/radare2/radare2.c (hunk 1 above)
Comment (radare) [reply binr/radare2/radare2.c]: but if you want to go for an env var at least dont reuse one that have a different meaning. use for example `R_DEBUG_DEBUGGER` in radare2.c to set the cfg.verbose variable

**Comment by radare** [comment]
ping

**Comment by pawlos** [comment]
Hi @radare I'll fix them during the weekend. I'm a bit busy ATM

**Comment by pawlos** [comment]
I've pushed some changes with using dbg.verbose. I hope this is what was envisioned here. If not please let me know.

**Pull request merged**